esp/platform/txsummary.cpp

```diff
@@ -81,8 +97,41 @@ bool CTxSummary::set(const char* key, const char* value)
     return true;
 }
 
+CumulativeTimer* CTxSummary::getTimer(const char* name)
+{
+    if (!name || !*name)
+        return NULL;
+
+    synchronized block(m_sync);
```

Comment:
I like to make these boundaries as tight as possible, do we need to protect the instantiation of key?